components/eam/src/physics/cam/macrop_driver.F90

```diff
@@ -321,6 +330,14 @@ subroutine macrop_driver_init(pbuf2d)
     CC_ni_idx   = pbuf_get_index('CC_ni')
     CC_qlst_idx = pbuf_get_index('CC_qlst')
 
+    dlfzm_idx = pbuf_get_index('DLFZM')
+    difzm_idx = pbuf_get_index('DIFZM')
+    dsfzm_idx = pbuf_get_index('DSFZM', err)
+    dnlfzm_idx = pbuf_get_index('DNLFZM', err)
+    dnifzm_idx = pbuf_get_index('DNIFZM', err)
+    dnsfzm_idx = pbuf_get_index('DNSFZM', err)
+    if (dnlfzm_idx > 0) zmconv_microp = .true.
```

Comment:
I don't like the look of this. Why do we need a separate "zmconv_microp" flag when we already have the namelist variable "zm_microp"? And why are we setting zmconv_microp based on the existence of "dnlfzm"? The existence of "dnlfzm" in the pbuf depends directly on "zm_microp=.true.", so let's just use "zm_microp" directly and not muddy the waters with a separate flag (unless there's a good reason that I'm not seeing).

Reply:
"zm_microp" is now directly used in macrop_driver.F90.